api/app_analytics/mappers.py

```diff
@@ -119,21 +118,22 @@ def map_flux_tables_to_feature_evaluation_data(
 def map_input_labels_to_labels(input_labels: InputLabels) -> Labels:
     labels: Labels = {}
     for label, value in input_labels.items():
-        if label == "sdk_user_agent":
-            labels["user_agent"] = value
+        if label == "sdk_user_agent" or label == "user_agent":
+            labels["user_agent"] = map_user_agent_to_sdk_user_agent(value)
             continue
-        elif label == "user_agent":
-            # fastuaparser classifies unrecognized UAs as "Other" — assume these to
-            # represent server-side SDKs.
-            parsed_ua_string: str = parse_ua(value)
-            is_server_side_sdk = parsed_ua_string.startswith("Other")
-            # Skip browser SDKs that don't send the special header.
-            if not is_server_side_sdk:
-                continue
         labels[label] = value
     return labels
 
 
+def map_user_agent_to_sdk_user_agent(value: str) -> str:
+    sdk_name, sdk_version = value.split("/")
+    if versions := SDK_USER_AGENT_KNOWN_VERSIONS.get(sdk_name):
+        if sdk_version in versions:
+            return value
+        return f"{sdk_name}/unknown"
+    return "unknown/unknown"
```

Comment:
Adding to the [other comment](https://github.com/Flagsmith/flagsmith/pull/6000/files#r2308555234), this could be simplified to:


```suggestion
    sdk_name, sdk_version = value.split("/")
    if sdk_name not in SDK_USER_AGENT_KNOWN_VERSIONS:
        sdk_name = "unknown"
    if not RE_SEMVER.match(sdk_version):
        sdk_version = "unknown"
    return f"{sdk_name}/{sdk_version}"
```

For `RE_SEMVER`, refer to semver.org [regex notes](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string).

Reply:
See https://github.com/Flagsmith/flagsmith/pull/6000/files#r2310648153.